lib/ansible/modules/storage/netapp/na_ontap_broadcast_domain.py

```diff
@@ -0,0 +1,253 @@
+        domain_obj.add_new_child("broadcast-domain", self.broadcast_domain)
+        if self.ipspace:
+            domain_obj.add_new_child("ipspace", self.ipspace)
+        try:
+            self.server.invoke_successfully(domain_obj, True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error deleting broadcast domain %s: %s' %
+                                  (self.broadcast_domain, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def modify_broadcast_domain(self):
+        """
+        Modifies ipspace and mtu options of a broadcast domain
+        """
+        domain_obj = netapp_utils.zapi.NaElement('net-port-broadcast-domain-modify')
+        domain_obj.add_new_child("broadcast-domain", self.broadcast_domain)
+        if self.ipspace:
+            domain_obj.add_new_child("ipspace", self.ipspace)
+        if self.mtu:
+            domain_obj.add_new_child("mtu", self.mtu)
+        try:
+            self.server.invoke_successfully(domain_obj, True)
+        except netapp_utils.zapi.NaApiError as error:
+            self.module.fail_json(msg='Error modifying broadcast domain %s: %s' %
+                                  (self.broadcast_domain, to_native(error)),
+                                  exception=traceback.format_exc())
+
+    def apply(self):
+        """
+        Run Module based on play book
+        """
+        changed = False
+        broadcast_domain_details = self.get_broadcast_domain()
+        broadcast_domain_exists = False
+        results = netapp_utils.get_cserver(self.server)
+        cserver = netapp_utils.setup_na_ontap_zapi(module=self.module, vserver=results)
+        netapp_utils.ems_log_event("na_ontap_broadcast_domain", cserver)
+        if broadcast_domain_details:
+            broadcast_domain_exists = True
+            if self.state == 'absent':  # delete
+                changed = True
+            elif self.state == 'present':  # modify
+                if (self.mtu and self.mtu != broadcast_domain_details['mtu']) or \
+                   (self.ipspace and self.ipspace != broadcast_domain_details['ipspace']):
+                    changed = True
+        else:
+            if self.state == 'present':  # create
+                changed = True
+        if changed:
+            if self.module.check_mode:
+                pass
+            else:
+                if self.state == 'present':  # execute create
+                    if not broadcast_domain_exists:
+                        self.create_broadcast_domain()
+                    else:  # execute modify
+                        self.modify_broadcast_domain()
+                elif self.state == 'absent':  # execute delete
+                    self.delete_broadcast_domain()
+        self.module.exit_json(changed=changed)
+
+
+def main():
+    """
+    Creates the NetApp Ontap Net Route object and runs the correct play task
```

Comment:
Should it be broadcast domain?

Reply:
Ya looks like this was copied and pasted from another file let me fix that.